### PR TITLE
💰 Add Update Working Group Budget proposal

### LINKED ISSUE #1857: Update Working Group Budget
See: https://joystream.gitbook.io/joystream-handbook/governance/proposals#update-working-group-budget

### PR DESCRIPTION
**Pull request by RafalPloszka**
Closes #1857

### CODE REVIEW

File: packages/ui/src/common/hooks/useNumberInput.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import BN from 'bn.js'
 import { useEffect, useState } from 'react'
 
-const NUMBER_REGEX = /^\d*(\.\d*)?$/
+const NUMBER_REGEX = /^-?\d*(\.\d*)?$/
```
[the remark this answers is not in the document]
Comment (jodator) [reply packages/ui/src/common/hooks/useNumberInput.ts:4]: For this ticket I would consider:

*   using different UI. For instance a flip-switch toggle to select negative/positive (or tu "enable" negative value"
*   adding optional config to the `useNumberInput` that will enable negative values.

The former is a bit quirkier on the UI side, but maybe not so much in terms of constructing proper proposal parameters.

The latter has broader consequences on code: requires properties refactoring across the whole app and adding tests to the `useNumberInput()` tests with the new behavior.

TL:DR: I'd go with a toggle to "enable" decreasing value.



image

**Review by jodator — CHANGES_REQUESTED** [review]
See my comments in the PR. Unfortunatelly using negative values in the input has few downsides:

- breaks the app as the values passed to the extrinsic are invalid
- the number input hook is used elsewhere and it in 99% must be positive value

I added my idea abut changing the form, but other ideas area welcome :)

File: packages/ui/src/proposals/modals/AddNewProposal/components/SpecificParameters/UpdateWorkingGroupBudget.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  const { group } = useWorkingGroup({ name: groupId })
+
+  const isDisabled = !group
+
+  useEffect(() => {
+    setBudgetUpdate(new BN(amount))
+  }, [amount])
+  useEffect(() => {
+    setBudgetUpdate(BN_ZERO)
+  }, [groupId])
```
[the remark this answers is not in the document]
Comment (RafalPloszka) [reply packages/ui/src/proposals/modals/AddNewProposal/components/SpecificParameters/UpdateWorkingGroupBudget.tsx:45]: I think it is correct now because it's not group budget but the update only. So when another group is selected the update input value should change to zero and then the user can decide whether to increase or decrease the budget.
The current group budget is handled elsewhere.

**Comment by jodator** [comment]
@RafalPloszka See my commits – unfortunately, there was an error while creating this proposal. The budget and working group params were switched in the tuple.

Also, the kind switch was also inverted and never changed.

**Review by jodator — APPROVED** [review]
